signer/src/error.rs

```diff
@@ -25,6 +25,10 @@ pub enum Error {
     #[error("idpack segmenter error: {0}")]
     IdPackSegmenter(#[from] sbtc::idpack::SegmenterError),
 
+    /// IdPack segments decode error
+    #[error("idpack segments decode error: {0}")]
+    IdPackDecodeError(#[source] sbtc::idpack::DecodeError),
```

Comment:
Suggestion: I'd just call this `IdPackDecode` as its implicitly an `Error`, and probably use `#[from]` as this is is a generic variant with no added context. But I know @djordon has his feelings on that 😅

Reply:
😄 I'm okay with `#[from]` for our internal types, but `#[source]` is the better default. So I'm okay either way. I missed the `Error` suffix though, that one should go.

Reply:
Going for it for the symmetry with `IdPackSegmenter`